{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import glob\n",
    "import scipy\n",
    "import matplotlib\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "from collections import defaultdict\n",
    "from matplotlib import pyplot as plt\n",
    "from matplotlib.backends.backend_pdf import PdfPages\n",
    "from shutil import copyfile\n",
    "\n",
    "from tqdm import tqdm\n",
    "\n",
    "%matplotlib inline\n",
    "sns.set_style('whitegrid')\n",
    "pd.set_option('display.max_rows', 100)\n",
    "matplotlib.rcParams['ps.fonttype'] = 42\n",
    "matplotlib.rcParams['pdf.fonttype'] = 42\n",
    "pd.set_option('display.max_columns', 100)\n",
    "\n",
    "sys.path.append('/home/mattolm/Bio_scripts/')\n",
    "\n",
    "import shutil\n",
    "from IPython.display import display, HTML\n",
    "\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://github.com/alexcritschristoph/wastewater_sarscov2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Links to input data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "PROBLEMATIC_SITES_LOCATION = '/groups/banfield/projects/industrial/wastewater_bayarea/2020/GithubRepo/wastewater_sarscov2/public_data/problematic_sites_sarsCov2.vcf'\n",
    "EXLUDE_STRAINS_LOCATION = '/groups/banfield/projects/industrial/wastewater_bayarea/2020/GithubRepo/wastewater_sarscov2/public_data/exclude.txt'\n",
    "\n",
    "IS_PROFILE_FOLDER_LOCATION = '/groups/banfield/projects/industrial/wastewater_bayarea/2020/GithubRepo/wastewater_sarscov2/data/wastewater/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Links to non-github input data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "MSA_LOCATION = '/groups/banfield/projects/industrial/wastewater_bayarea/2020/combined_good/patient_aug/msa_0824.fasta'\n",
    "METADATA_LOCATION = '/groups/banfield/projects/industrial/wastewater_bayarea/2020/combined_good/patient_aug/metadata_2020-08-24_12-12.tsv'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Links to where data will be saved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "INTRAPATIENT_SNVS_SAVE_LOCATION = '/groups/banfield/projects/industrial/wastewater_bayarea/2020/GithubRepo/wastewater_sarscov2/tables/Interpatient_SNVs_v2.csv.gz'\n",
    "INTRAPATIENT_SNVS_SNPLEVEL_SAVE_LOCATION ='/groups/banfield/projects/industrial/wastewater_bayarea/2020/GithubRepo/wastewater_sarscov2/tables/Interpatient_SNVs_SNPlevel_v2.csv.gz'\n",
    "\n",
    "WASTEWATER_SNVS_SAVELOC = '/groups/banfield/projects/industrial/wastewater_bayarea/2020/GithubRepo/wastewater_sarscov2/tables/Wastewater_SNVs_v2.csv'\n",
    "WASTEWATER_ABUND_SAVELOC = '/groups/banfield/projects/industrial/wastewater_bayarea/2020/GithubRepo/wastewater_sarscov2/tables/Wastewater_abundance_v2.csv'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load interpatient SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mattolm/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pandas/core/indexing.py:965: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[item] = s\n",
      "168it [00:00, 1047.62it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 30861 columns\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "30861it [11:28, 44.82it/s]\n"
     ]
    }
   ],
   "source": [
    "def load_SNPs_from_MSA(msa_loc, reference_id='hCoV-19/Wuhan/WIV04/2019|EPI_ISL_402124|2019-12-30|China', ignore=set()):\n",
    "    \"\"\"\n",
    "    Load SNPs from a multiple sequence alignment\n",
    "    \"\"\"\n",
    "    table = defaultdict(list)\n",
    "    \n",
    "    # Load the referece sequence\n",
    "    reference = None\n",
    "    for record in SeqIO.parse(msa_loc, 'fasta'):\n",
    "        if record.id == reference_id:\n",
    "            reference = record\n",
    "            break\n",
    "    if reference is None:\n",
    "        print(\"Cannot find sequence {0}\".format(reference_id))\n",
    "        raise Exception()\n",
    "        \n",
    "    # Load the rest of the sequences into RAM\n",
    "    record_ids = []\n",
    "    record_sequences = []\n",
    "    record_SNPs = defaultdict(list)\n",
    "    for record in SeqIO.parse(msa_loc, 'fasta'):\n",
    "        record_ids.append(record.description)\n",
    "        record_sequences.append(str(record.seq).upper())\n",
    "        \n",
    "    # Call SNPs\n",
    "    print(\"There are {0} columns\".format(len(str(reference.seq).upper())))\n",
    "    reference_position = 0\n",
    "    #for bases in tqdm(zip(str(reference.seq).upper(), *record_sequences)):\n",
    "    for column, ref_base in tqdm(enumerate(str(reference.seq).upper())):\n",
    "        if ref_base in ['A', 'C', 'G', 'T']:\n",
    "            if reference_position in ignore:\n",
    "                reference_position += 1\n",
    "                continue\n",
    "            \n",
    "            bases = [r[column] for r in record_sequences]\n",
    "            for i, base in enumerate(bases): # loop through the records and this column\n",
    "                if base in ['A', 'C', 'G', 'T']:\n",
    "                    if base != ref_base:\n",
    "                        record_SNPs[record_ids[i]].append((reference_position, column, base))\n",
    "                else:\n",
    "                    record_SNPs[record_ids[i]].append((reference_position, column, 'NA'))\n",
    "            reference_position += 1\n",
    "        else:\n",
    "            pass\n",
    "        \n",
    "    # Make table\n",
    "    table = defaultdict(list)\n",
    "    for record in record_ids:\n",
    "        SNPs = record_SNPs[record]\n",
    "        table['record'].append(record)\n",
    "        table['SNP_count'].append(len([True for s in SNPs if s[2] != 'NA']))\n",
    "        table['missing_count'].append(len([True for s in SNPs if s[2] == 'NA']))\n",
    "        table['SNP_IDs'].append(set(SNPs))\n",
    "        \n",
    "    ISdb = pd.DataFrame(table)\n",
    "\n",
    "    return ISdb\n",
    "\n",
    "# Load problem sites\n",
    "problem_sites = pd.read_csv(PROBLEMATIC_SITES_LOCATION, comment=\"#\", sep=\"\\t\")\n",
    "filter_sites = problem_sites.query(\"FILTER=='mask'\")\n",
    "filter_sites.loc[:,'POS'] = filter_sites['POS'] - 1\n",
    "filter_sites = set(filter_sites['POS'].tolist())\n",
    "\n",
    "IPdb = load_SNPs_from_MSA(MSA_LOCATION, ignore=filter_sites)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Incorporate metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9,064 strains are in the metadata but not in the MSA\n",
      "0 strains are in the MSA but not in the metadata\n"
     ]
    }
   ],
   "source": [
    "MEdb = pd.read_csv(METADATA_LOCATION, sep='\\t')\n",
    "\n",
    "exclude = []\n",
    "f = open(EXLUDE_STRAINS_LOCATION)\n",
    "for line in f.readlines():\n",
    "    if not line.startswith(\"#\") and line.strip() != '':\n",
    "        exclude.append(line.strip())\n",
    "\n",
    "IPMdb = IPdb.copy()\n",
    "IPMdb['strain'] = [x.replace('hCoV-19/', '').split('|')[0] for x in IPMdb['record']]\n",
    "IPMdb = pd.merge(IPMdb, MEdb, on='strain', how='outer')\n",
    "IPMdb = IPMdb[IPMdb['host'] == 'Human']\n",
    "IPMdb = IPMdb[~IPMdb['strain'].isin(exclude)]\n",
    "\n",
    "print(\"{0:,} strains are in the metadata but not in the MSA\".format(len(IPMdb[IPMdb['record'].isna()])))\n",
    "print(\"{0:,} strains are in the MSA but not in the metadata\".format(len(IPMdb[IPMdb['region'].isna()])))\n",
    "\n",
    "Idb = IPMdb[~IPMdb['record'].isna()]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "Idb.to_csv(INTRAPATIENT_SNVS_SAVE_LOCATION, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make a SNV-level table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 21,554 unique SNPs in this dataset\n"
     ]
    }
   ],
   "source": [
    "ALL_SNPS = set(x for x in set.union(*Idb['SNP_IDs'].tolist()) if x[2] != 'NA')\n",
    "print(\"There are {0:,} unique SNPs in this dataset\".format(len(ALL_SNPS)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 21554/21554 [08:37<00:00, 41.65it/s]\n"
     ]
    }
   ],
   "source": [
    "table = defaultdict(list)\n",
    "for SNP in tqdm(ALL_SNPS):\n",
    "    db = Idb[[SNP in x for x in Idb['SNP_IDs']]]\n",
    "    \n",
    "    table['SNP'].append(str(SNP))\n",
    "    table['number_samples'].append(len(db))\n",
    "    table['strains'].append(set(db['strain'].tolist()))\n",
    "\n",
    "Vdb = pd.DataFrame(table)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "Vdb.to_csv(INTRAPATIENT_SNVS_SNPLEVEL_SAVE_LOCATION, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make wastewater tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome</th>\n",
       "      <th>detected_scaffolds</th>\n",
       "      <th>true_scaffolds</th>\n",
       "      <th>true_length</th>\n",
       "      <th>SNPs</th>\n",
       "      <th>BiAllelic_SNPs</th>\n",
       "      <th>MultiAllelic_SNPs</th>\n",
       "      <th>consensus_SNPs</th>\n",
       "      <th>population_SNPs</th>\n",
       "      <th>breadth</th>\n",
       "      <th>coverage</th>\n",
       "      <th>std_cov</th>\n",
       "      <th>mean_clonality</th>\n",
       "      <th>rarefied_mean_microdiversity</th>\n",
       "      <th>conANI</th>\n",
       "      <th>popANI</th>\n",
       "      <th>unmaskedBreadth</th>\n",
       "      <th>expected_breadth</th>\n",
       "      <th>sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>all_scaffolds</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>29840</td>\n",
       "      <td>70</td>\n",
       "      <td>64</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>0.999129</td>\n",
       "      <td>33.796247</td>\n",
       "      <td>8.191150</td>\n",
       "      <td>0.997073</td>\n",
       "      <td>0.002109</td>\n",
       "      <td>0.999765</td>\n",
       "      <td>0.999799</td>\n",
       "      <td>0.999062</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>6_30_S_MOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>all_scaffolds</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>29840</td>\n",
       "      <td>42</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>9</td>\n",
       "      <td>0.999765</td>\n",
       "      <td>88.467225</td>\n",
       "      <td>16.853628</td>\n",
       "      <td>0.997316</td>\n",
       "      <td>0.002547</td>\n",
       "      <td>0.999531</td>\n",
       "      <td>0.999698</td>\n",
       "      <td>0.999296</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>MR_7_1_MOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>all_scaffolds</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>29840</td>\n",
       "      <td>37</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>0.995912</td>\n",
       "      <td>8.407775</td>\n",
       "      <td>5.715350</td>\n",
       "      <td>0.995928</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.999786</td>\n",
       "      <td>0.999822</td>\n",
       "      <td>0.940885</td>\n",
       "      <td>0.999403</td>\n",
       "      <td>5_28_S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>all_scaffolds</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>29840</td>\n",
       "      <td>42</td>\n",
       "      <td>34</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>0.999397</td>\n",
       "      <td>32.567828</td>\n",
       "      <td>7.868926</td>\n",
       "      <td>0.997842</td>\n",
       "      <td>0.001941</td>\n",
       "      <td>0.999631</td>\n",
       "      <td>0.999732</td>\n",
       "      <td>0.999162</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>6_30_S_COL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>all_scaffolds</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>29840</td>\n",
       "      <td>16</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>0.999497</td>\n",
       "      <td>107.051642</td>\n",
       "      <td>28.055301</td>\n",
       "      <td>0.998380</td>\n",
       "      <td>0.001509</td>\n",
       "      <td>0.999732</td>\n",
       "      <td>0.999799</td>\n",
       "      <td>0.998727</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>5_19_S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>all_scaffolds</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>29840</td>\n",
       "      <td>73</td>\n",
       "      <td>65</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0.999229</td>\n",
       "      <td>13.037500</td>\n",
       "      <td>8.045137</td>\n",
       "      <td>0.997008</td>\n",
       "      <td>0.007359</td>\n",
       "      <td>0.999525</td>\n",
       "      <td>0.999729</td>\n",
       "      <td>0.988673</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>SQ_7_1_MOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>all_scaffolds</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>29840</td>\n",
       "      <td>31</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>0.999129</td>\n",
       "      <td>10.832574</td>\n",
       "      <td>4.921677</td>\n",
       "      <td>0.997967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.999796</td>\n",
       "      <td>0.999864</td>\n",
       "      <td>0.985757</td>\n",
       "      <td>0.999930</td>\n",
       "      <td>5_19_S_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>all_scaffolds</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>29840</td>\n",
       "      <td>25</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>0.994437</td>\n",
       "      <td>7.075737</td>\n",
       "      <td>4.307690</td>\n",
       "      <td>0.996682</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.999813</td>\n",
       "      <td>0.999925</td>\n",
       "      <td>0.897420</td>\n",
       "      <td>0.998065</td>\n",
       "      <td>6_09_S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          genome  detected_scaffolds  true_scaffolds  true_length  SNPs  \\\n",
       "0  all_scaffolds                   1               1        29840    70   \n",
       "1  all_scaffolds                   1               1        29840    42   \n",
       "2  all_scaffolds                   1               1        29840    37   \n",
       "3  all_scaffolds                   1               1        29840    42   \n",
       "4  all_scaffolds                   1               1        29840    16   \n",
       "5  all_scaffolds                   1               1        29840    73   \n",
       "6  all_scaffolds                   1               1        29840    31   \n",
       "7  all_scaffolds                   1               1        29840    25   \n",
       "\n",
       "   BiAllelic_SNPs  MultiAllelic_SNPs  consensus_SNPs  population_SNPs  \\\n",
       "0              64                  0               7                6   \n",
       "1              33                  0              14                9   \n",
       "2              32                  0               6                5   \n",
       "3              34                  0              11                8   \n",
       "4              10                  0               8                6   \n",
       "5              65                  0              14                8   \n",
       "6              27                  0               6                4   \n",
       "7              23                  0               5                2   \n",
       "\n",
       "    breadth    coverage    std_cov  mean_clonality  \\\n",
       "0  0.999129   33.796247   8.191150        0.997073   \n",
       "1  0.999765   88.467225  16.853628        0.997316   \n",
       "2  0.995912    8.407775   5.715350        0.995928   \n",
       "3  0.999397   32.567828   7.868926        0.997842   \n",
       "4  0.999497  107.051642  28.055301        0.998380   \n",
       "5  0.999229   13.037500   8.045137        0.997008   \n",
       "6  0.999129   10.832574   4.921677        0.997967   \n",
       "7  0.994437    7.075737   4.307690        0.996682   \n",
       "\n",
       "   rarefied_mean_microdiversity    conANI    popANI  unmaskedBreadth  \\\n",
       "0                      0.002109  0.999765  0.999799         0.999062   \n",
       "1                      0.002547  0.999531  0.999698         0.999296   \n",
       "2                      0.000000  0.999786  0.999822         0.940885   \n",
       "3                      0.001941  0.999631  0.999732         0.999162   \n",
       "4                      0.001509  0.999732  0.999799         0.998727   \n",
       "5                      0.007359  0.999525  0.999729         0.988673   \n",
       "6                      0.000000  0.999796  0.999864         0.985757   \n",
       "7                      0.000000  0.999813  0.999925         0.897420   \n",
       "\n",
       "   expected_breadth      sample  \n",
       "0          1.000000  6_30_S_MOS  \n",
       "1          1.000000  MR_7_1_MOS  \n",
       "2          0.999403      5_28_S  \n",
       "3          1.000000  6_30_S_COL  \n",
       "4          1.000000      5_19_S  \n",
       "5          0.999990  SQ_7_1_MOS  \n",
       "6          0.999930    5_19_S_2  \n",
       "7          0.998065      6_09_S  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Adb = []\n",
    "for fn in glob.glob(IS_PROFILE_FOLDER_LOCATION + '*/output/*genomeWide_scaffold_info.tsv'):\n",
    "    adb = pd.read_csv(fn, sep='\\t')\n",
    "    adb['sample'] = os.path.basename(fn).split('_genomeWide')[0]\n",
    "    Adb.append(adb)\n",
    "Adb = pd.concat(Adb).reset_index(drop=True)\n",
    "Adb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>refBase</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>T</th>\n",
       "      <th>G</th>\n",
       "      <th>conBase</th>\n",
       "      <th>varBase</th>\n",
       "      <th>allele_count</th>\n",
       "      <th>cryptic</th>\n",
       "      <th>baseCoverage</th>\n",
       "      <th>varFreq</th>\n",
       "      <th>conFreq</th>\n",
       "      <th>refFreq</th>\n",
       "      <th>sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>33</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>8</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>6_30_S_MOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>34</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>7</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>6_30_S_MOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>35</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>9</td>\n",
       "      <td>0.222222</td>\n",
       "      <td>0.777778</td>\n",
       "      <td>0.777778</td>\n",
       "      <td>6_30_S_MOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>240</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>32</td>\n",
       "      <td>0.031250</td>\n",
       "      <td>0.968750</td>\n",
       "      <td>0.031250</td>\n",
       "      <td>6_30_S_MOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>1025</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>26</td>\n",
       "      <td>0.076923</td>\n",
       "      <td>0.923077</td>\n",
       "      <td>0.923077</td>\n",
       "      <td>6_30_S_MOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>28377</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>28</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>6_09_S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>28880</td>\n",
       "      <td>G</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>25</td>\n",
       "      <td>0.280000</td>\n",
       "      <td>0.720000</td>\n",
       "      <td>0.280000</td>\n",
       "      <td>6_09_S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>28881</td>\n",
       "      <td>G</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>25</td>\n",
       "      <td>0.280000</td>\n",
       "      <td>0.720000</td>\n",
       "      <td>0.280000</td>\n",
       "      <td>6_09_S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>28882</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>25</td>\n",
       "      <td>0.280000</td>\n",
       "      <td>0.720000</td>\n",
       "      <td>0.280000</td>\n",
       "      <td>6_09_S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...</td>\n",
       "      <td>29448</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>14</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>6_09_S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>336 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              scaffold  position refBase   A  \\\n",
       "0    hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...        33       A   6   \n",
       "1    hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...        34       A   5   \n",
       "2    hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...        35       C   0   \n",
       "3    hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...       240       C   0   \n",
       "4    hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...      1025       C   0   \n",
       "..                                                 ...       ...     ...  ..   \n",
       "331  hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...     28377       G   0   \n",
       "332  hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...     28880       G  18   \n",
       "333  hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...     28881       G  18   \n",
       "334  hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...     28882       G   0   \n",
       "335  hCoV-19_Wuhan_WIV04_2019_EPI_ISL_402124_2019-1...     29448       G   0   \n",
       "\n",
       "      C   T   G conBase varBase  allele_count  cryptic  baseCoverage  \\\n",
       "0     0   2   0       A       T             2    False             8   \n",
       "1     0   2   0       A       T             2    False             7   \n",
       "2     7   2   0       C       T             2    False             9   \n",
       "3     1  31   0       T       C             1    False            32   \n",
       "4    24   2   0       C       T             2    False            26   \n",
       "..   ..  ..  ..     ...     ...           ...      ...           ...   \n",
       "331   0   8  20       G       T             2    False            28   \n",
       "332   0   0   7       A       G             2    False            25   \n",
       "333   0   0   7       A       G             2    False            25   \n",
       "334  18   0   7       C       G             2    False            25   \n",
       "335   0   2  12       G       T             2    False            14   \n",
       "\n",
       "      varFreq   conFreq   refFreq      sample  \n",
       "0    0.250000  0.750000  0.750000  6_30_S_MOS  \n",
       "1    0.285714  0.714286  0.714286  6_30_S_MOS  \n",
       "2    0.222222  0.777778  0.777778  6_30_S_MOS  \n",
       "3    0.031250  0.968750  0.031250  6_30_S_MOS  \n",
       "4    0.076923  0.923077  0.923077  6_30_S_MOS  \n",
       "..        ...       ...       ...         ...  \n",
       "331  0.285714  0.714286  0.714286      6_09_S  \n",
       "332  0.280000  0.720000  0.280000      6_09_S  \n",
       "333  0.280000  0.720000  0.280000      6_09_S  \n",
       "334  0.280000  0.720000  0.280000      6_09_S  \n",
       "335  0.142857  0.857143  0.857143      6_09_S  \n",
       "\n",
       "[336 rows x 16 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "al = []\n",
    "for fn in glob.glob(IS_PROFILE_FOLDER_LOCATION + '*/output/*SNVs.tsv'):\n",
    "    adb = pd.read_csv(fn, sep='\\t')\n",
    "    adb['sample'] = os.path.basename(fn).split('_SNVs')[0]\n",
    "    al.append(adb)\n",
    "Sdb = pd.concat(al).reset_index(drop=True)\n",
    "Sdb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_SNP(row):\n",
    "    if row['allele_count'] == 1:\n",
    "        return (row['position'], row['conBase'])\n",
    "    elif row['allele_count'] == 2:\n",
    "        if row['conBase'] != row['refBase']:\n",
    "            return (row['position'], row['conBase'])\n",
    "        else:\n",
    "            assert row['varBase'] != row['refBase'], str(row)\n",
    "            return (row['position'], row['varBase'])\n",
    "    else:\n",
    "        raise Exception()\n",
    "        \n",
    "def key_freq(row):\n",
    "    return row[row['key'][1]] / row['baseCoverage']\n",
    "\n",
    "Sdb['key'] = Sdb.apply(parse_SNP, axis=1)\n",
    "Sdb['key_freq'] = Sdb.apply(key_freq, axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "Sdb.to_csv(WASTEWATER_SNVS_SAVELOC, index=False)\n",
    "Adb.to_csv(WASTEWATER_ABUND_SAVELOC, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
